remotes/docker/scope.go

```diff
@@ -51,21 +51,31 @@ func contextWithRepositoryScope(ctx context.Context, refspec reference.Spec, pus
 	if err != nil {
 		return nil, err
 	}
-	return context.WithValue(ctx, tokenScopesKey{}, []string{s}), nil
+	return contextWithScope(ctx, s), nil
 }
 
-// contextWithAppendPullRepositoryScope is used to append repository pull
-// scope into existing scopes indexed by the tokenScopesKey{}.
-func contextWithAppendPullRepositoryScope(ctx context.Context, repo string) context.Context {
+func contextWithScope(ctx context.Context, scope string) context.Context {
 	var scopes []string
-
 	if v := ctx.Value(tokenScopesKey{}); v != nil {
-		scopes = append(scopes, v.([]string)...)
+		scopes = v.([]string)
+		scopes = append(scopes, scope)
+	} else {
+		scopes = []string{scope}
 	}
-	scopes = append(scopes, fmt.Sprintf("repository:%s:pull", repo))
 	return context.WithValue(ctx, tokenScopesKey{}, scopes)
 }
 
+// contextWithAppendPullRepositoryScope is used to append repository pull
+// scope into existing scopes indexed by the tokenScopesKey{}.
+func contextWithAppendPullRepositoryScope(ctx context.Context, repo string) context.Context {
+	return contextWithScope(ctx, fmt.Sprintf("repository:%s:pull", repo))
+}
+
+// ContextWithScope appends a custom registry auth scope to the context.
+func ContextWithScope(ctx context.Context, scope string) context.Context {
```

Comment:
For these context modifiers, usually we just follow an abbreviated naming convention `WithScope` since context is implied through the signature and it is a fairly familiar pattern.

Reply:
fixed